decision-records/2024-06-25--extra-tempo-receivers.md

```diff
@@ -0,0 +1,59 @@
+# Extra tempo receivers
+**Date:** 2024-06-25<br/>
+**Authors:** @PietroPasotti, @mmkay, @michaeldmitry
+
+## Context and Problem Statement
+Charms seeking to push traces to Tempo will relate to the Tempo charm over a `tracing` relation. [The protocol](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/tracing/v2) specifies that the requirer charm needs to request a list of the tracing protocols it intends to use to send spans to Tempo (e.g: `otlp_http`, `jaeger`). Then, as a response, the provider Tempo will share for each requested protocol (so long as it supports it), an endpoint to which workloads/charms can push traces.
+Tempo only enables receivers for the protocols it actually needs to satisfy all incoming requests. Therefore, if a user wants to send traces to Tempo over a protocol that no charm is currently requesting via the `tracing` interface, there needs to be a way to configure the Tempo charm to enable that receiver regardless of whether there is a `tracing` integration asking for it the regular way. 
```

Comment:
It is also our choice (although a rather obvious one) to support uncharmed tracing. We might want to mention that we decided to allow uncharmed applications to send traces as well.